File: Takina-artist-service/src/main/java/com/takina/artist/service/entities/Artista.java
package com.takina.artist.service.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(
        name="artistas",
        uniqueConstraints = {
                @UniqueConstraint(name="artistas_nombre_unique", columnNames = "nombre")
        }
)
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Artista {

    // ---------------------

    @Id
    @GeneratedValue(
            strategy = GenerationType.IDENTITY
    )
    @Column(
            name="id",
            updatable = false
    )
    private Long id;

    // ---------------------

    @Column(
            name = "nombre",
            nullable = false,
            columnDefinition = "VARCHAR(50)",
            length = 50
    )
    private String nombre;

    // ---------------------

    @Column(
            name = "biografia",
            nullable = false,
            columnDefinition = "VARCHAR(140)",
            length = 140
    )
    private String biografia;

    // ---------------------

    @Column(
            name = "foto_perfil",
            nullable = false,
            columnDefinition = "VARCHAR(50)",
            length = 50
    )
    private String fotoPerfil;

    // ---------------------

    @Column(
            name = "fotoPortada",
            nullable = false,
            columnDefinition = "VARCHAR(50)",
            length = 50
    )
    private String fotoPortada;

    // --------------------

    @Column(
            name = "departamento",
            nullable = false,
            columnDefinition = "VARCHAR(20)",
            length = 20
    )
    private String departamento;

    // --------------------

    @Column(
            name = "genero",
            nullable = false,
            columnDefinition = "VARCHAR(20)",
            length = 20
    )
    private String genero;

    // ---------------------

    @Column(
            name = "total_seguidores"
    )
    private Long totalSeguidores = 0L;

    // ---------------------

    @Column(
            name = "reproducciones_total"
    )
    private Long totalReproducciones = 0L;

    // --------------------
    // Relaciones con las tablas de base de datos

    // --------------------
    // Administradores (One Usuario - Many Administrador - One Artista)
    @OneToMany(
            mappedBy = "artista",
            cascade = {CascadeType.PERSIST, CascadeType.REMOVE},
            fetch = FetchType.LAZY
    )
    private List<Administrador> administradores = new ArrayList<>();

    @OneToMany(
            mappedBy = "artista",
            cascade = {CascadeType.PERSIST, CascadeType.REMOVE},
            fetch = FetchType.LAZY
    )
    private List<Seguidor> seguidores = new ArrayList<>();


}

File: src/app/models/artista.ts
import {  ReqArtista } from "./artistaResponse";

export class Artista{

    static artistaFromJson( obj: ReqArtista){
        return new Artista(
            obj['biografia'],
            obj['departamento'], 
            obj['fotoPerfil'],
            obj['fotoPortada'], 
            obj['genero'],       
            obj['id'],  
            obj['nombre'],   
            obj['totalReproducciones'],
            obj['totalSeguidores'],

        )
    }

    constructor(
        public biografia:           string,
        public departamento:        string,
        public fotoPerfil:          string,
        public fotoPortada:         string,
        public genero:              string,
        public id:                  number,
        public nombre:              string,
        public totalReproducciones: number,
        public totalSeguidores:     number,
    ){}

    
}

export class CreateArtista {
	constructor(
		public biografia: string,
		public departamento: string,
		public fotoPerfil: string,
		public fotoPortada: string,
		public genero: string,
		public nombre: string,
		public usuarioId: number
	) {}
}

export class Administrador{
    constructor(
        public fechaRegistro: string,
        public nivel:         string,
        public usuarioNombre: string,
    ){}
}

export class Proyecto{
    constructor(
        public fecha:       string,
        public id:          number,
        public lanzamiento: string,
        public nombre:      string,
    ){}
}

export class Seguidor{
    constructor(
        public fecha:        string,
        public usuarioApodo: string,
        
    ){}
}